### ISSUE TITLE
"Pending Promise deallocated! This is usually a bug" when using When or Join

### ISSUE THREAD

**Issue by jonasman** [mxcl/PromiseKit#447]
Tried to use a when like this:

````swift
when(array_of_strings_Ids.map { (one_Id) -> Promise<()> in
            MyService.defaultService.getData(one_Id).then {
                (data) -> () in
                //do nothing, but i want to do something with the value
            } }  )
````

Using join gives me the same warning.

If i print the description of the when promise i will get this:
`Promise: UnsealedState: Pending with 0 handlers`

Is this normal?
Is there something wrong with the map create the when/join promise?

**Comment by mxcl** [comment]
Hmm, seems like a bug. If you have time can you construct a test project? Otherwise we will get round to it.

**Comment by jonasman** [comment]
I did make a project but wasn't able to reproduce the issue. I will keep trying until i can reproduce it. I must mention im using for this project PromiseKit 3.0.3

**Comment by seriyvolk83** [comment]
It still exists. If promise is not assigned to any variable then there is a change that it will be deallocated. However, sometimes the Promise object is not needed. Not sure why it is deallocated.

**Comment by mxcl** [comment]
We consider it an anti pattern to have pending promises that are deallocated. If you open a new ticket we can discuss it further.

**Comment by GertjanSmits** [comment]
I also get this, any known workaround for it? Kinda new to it all, not sure which one is deallocated, im chaining 3 promises.

**Comment by mxcl** [comment]
One of your promises is neither fulfilled or rejected. This will lead to bugs. Figure out which promise it is: trace all asynchronous routes and look for an `if` or something like that where the else is not calling either `fulfill` or `reject`.

**Comment by seriyvolk83** [comment]
@mxcl It's not the case. The initial promise was not saved into a variable and as a result the sequence of promises was interrupted due to deallocation because no object is pointing to the initial promise.

**Comment by mxcl** [comment]
@seriyvolk83 the `fulfill` and `reject` functions retain the promise, so unless you are using promises incorrectly it is not how you are thinking it is.

**Comment by GertjanSmits** [comment]
You're right, 1 of of my promises had a state heen it never called fulfilled/reject. Thanks!

**Comment by NikKovIos** [comment]
In my case it was, when i used `PromiseKit.wrap {}` on function, which returned nothing.
